torchmetrics/text/eed.py

```diff
@@ -0,0 +1,143 @@
+
+    Args:
+        language:
+            Language used in sentences. Only supports English (en) and Japanese (ja) for now. Defaults to en
+        return_sentence_level_score:
+            An indication of whether sentence-level EED is to be returned
+        compute_on_step:
+            Forward only calls ``update()`` and return None if this is set to False.
+        dist_sync_on_step:
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step.
+        process_group:
+            Specify the process group on which synchronization is called. default: None (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When ``None``, DDP
+            will be used to perform the allgather
+        alpha:
+            optimal jump penalty, penalty for jumps between characters
+        rho:
+            coverage cost, penalty for repetition of characters
+        deletion:
+            penalty for deletion of character
+        insertion:
+            penalty for insertion or substitution of character
+
+    Returns:
+        Extended edit distance score as a tensor
+
+    Example:
+        >>> from torchmetrics.text import EED
+        >>> reference_corpus = ["this is the reference", "here is another one"]
+        >>> hypothesis_corpus = ["this is the prediction", "here is an other sample"]
+        >>> metric = EED()
+        >>> metric(reference_corpus=reference_corpus, hypothesis_corpus=hypothesis_corpus)
+        tensor(0.3204)
+
+    References:
+        [1] P. Stanchev, W. Wang, and H. Ney, “EED: Extended Edit Distance Measure for Machine Translation”, submitted
+        to WMT 2019. `EED`_
+    """
+
+    scores: Tensor
+    total_num_sentences: Tensor
+    higher_is_better: False
+    is_differentiable: False
+
+    def __init__(
+        self,
+        language: Literal["en", "ja"] = "en",
+        return_sentence_level_score: bool = False,
+        compute_on_step: bool = True,
+        dist_sync_on_step: bool = False,
+        process_group: Optional[Any] = None,
+        dist_sync_fn: Callable = None,
+        alpha: float = 2.0,
+        rho: float = 0.3,
+        deletion: float = 0.2,
+        insertion: float = 1.0,
+    ):
+        super().__init__(
+            compute_on_step=compute_on_step,
+            dist_sync_on_step=dist_sync_on_step,
+            process_group=process_group,
+            dist_sync_fn=dist_sync_fn,
+        )
+
+        if language not in ("en", "ja"):
+            raise ValueError(f"Expected argument `language` to either be `en` or `ja` but got {language}")
+        self.language: Literal["en", "ja"] = language
+        self.return_sentence_level_score = return_sentence_level_score
+        self.sentence_eed = []
+
+        self.alpha = alpha
+        self.rho = rho
+        self.deletion = deletion
+        self.insertion = insertion
```

Comment:
Please add input validation for these arguments. I guess all should be positive floats

Reply:
ok